src/test/java/core/basesyntax/service/impl/CsvFormatReportGeneratorTest.java

```diff
@@ -0,0 +1,69 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import core.basesyntax.dao.FruitStorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverterService;
+import core.basesyntax.service.FileReaderService;
+import core.basesyntax.service.ReportGenerator;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.service.impl.operation.BalanceOperation;
+import core.basesyntax.service.impl.operation.OperationHandler;
+import core.basesyntax.service.impl.operation.PurchaseOperation;
+import core.basesyntax.service.impl.operation.ReturnOperation;
+import core.basesyntax.service.impl.operation.SupplyOperation;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.OperationStrategyImpl;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class CsvFormatReportGeneratorTest {
+    private static final String INPUT_FILE_PATH = "src/test/resources/reportToRead.csv";
+    private static Storage storage;
+    private static FruitStorageDaoImpl fruitStorageDao;
+    private static ReportGenerator csvFormatReportGenerator;
+    private static FileReaderService fileReader;
+    private static DataConverterService dataConverter;
+    private static OperationStrategy operationStrategy;
+    private static Map<FruitTransaction.Operation, OperationHandler> OPERATION_HANDLERS;
+    private static ShopService shopService;
```

Comment:
Why did you use a bunch of your dependencies and related classes here? Unit tests should be as much isolated as possible. Try manually adding data to your storage and using the least additional classes/services.

Reply:
fix the same issue in all places